assets/js/base/components/price-slider/index.tsx

```diff
@@ -471,24 +471,32 @@ const PriceSlider = ( {
 						/>
 					</div>
 				) }
-			{ ! isUpdating && (
+			{
 				<div className="wc-block-components-price-slider__actions">
-					{ ( minPrice !== minConstraint ||
-						maxPrice !== maxConstraint ) && (
-						<FilterResetButton
-							onClick={ () => {
-								onChange( [ minConstraint, maxConstraint ] );
-								debouncedUpdateQuery();
-							} }
-							screenReaderLabel={ __(
-								'Reset price filter',
-								'woo-gutenberg-products-block'
-							) }
-						/>
-					) }
+					{ ! isUpdating &&
+						( minPrice !== minConstraint ||
+							maxPrice !== maxConstraint ) && (
+							<FilterResetButton
+								onClick={ () => {
+									onChange( [
+										minConstraint,
+										maxConstraint,
+									] );
+									debouncedUpdateQuery();
+								} }
+								screenReaderLabel={ __(
+									'Reset price filter',
+									'woo-gutenberg-products-block'
+								) }
+							/>
+						) }
 					{ showFilterButton && (
 						<FilterSubmitButton
-							className="wc-block-price-filter__button wc-block-components-price-slider__button"
+							className={ classnames(
+								'wc-block-price-filter__button',
+								'wc-block-components-price-slider__button',
+								{ 'show-loading-state': isUpdating }
```

Comment:
Instead of setting the `show-loading-state` class in each filter block, could we pass the `isUpdating` prop to `FilterSubmitButton` and handle the styling inside the component? That way we don't have to repeat the same styling in the _Apply_ button of each filter block.

Reply:
Good idea 👍